CRM/Report/Form/Event/Income.php

```diff
@@ -111,18 +111,27 @@ public function buildEventReport($eventIDs) {
 
     $groupBy = CRM_Contact_BAO_Query::getGroupByFromSelectColumns($select, 'civicrm_event.id');
     $sql = "
-            SELECT  " . implode(', ', $select) . ",
-                    SUM(civicrm_participant.fee_amount) as total,
-                    COUNT(civicrm_participant.id)       as participant
-
-            FROM       civicrm_event
-            LEFT JOIN  civicrm_option_value
-                   ON  ( civicrm_event.event_type_id = civicrm_option_value.value AND
-                         civicrm_option_value.option_group_id = {$optionGroupId} )
-            LEFT JOIN  civicrm_participant ON ( civicrm_event.id = civicrm_participant.event_id
-                       {$activeParticipantClause} AND civicrm_participant.is_test  = 0 )
-
-            WHERE      civicrm_event.id IN( {$eventID}) {$groupBy}";
+      SELECT " . implode(', ', $select) . ",
+        SUM(civicrm_participant.fee_amount) as total,
+        COUNT(civicrm_participant.id) as participant
+      FROM civicrm_event
+      LEFT JOIN civicrm_option_value
+        ON civicrm_event.event_type_id = civicrm_option_value.value 
+        AND civicrm_option_value.option_group_id = {$optionGroupId}
+      LEFT JOIN (
+        SELECT civicrm_participant.*
+        FROM civicrm_participant
+        JOIN civicrm_contact
+          ON civicrm_participant.contact_id = civicrm_contact.id
+          AND civicrm_contact.is_deleted = 0
+        WHERE civicrm_participant.event_id IN ({$eventID})
+          {$activeParticipantClause}
+          AND civicrm_participant.is_test = 0
+        ) civicrm_participant
+        ON civicrm_participant.event_id = civicrm_event.id
```

Comment:
Just worried since civicrm_participant.event_id is not indexed and might impact performance?

How about this

```
LEFT JOIN  civicrm_participant ON ( civicrm_event.id = civicrm_participant.event_id
                       {$activeParticipantClause} AND civicrm_participant.is_test  = 0 )
LEFT JOIN civicrm_contact
          ON civicrm_participant.contact_id = civicrm_contact.id 
WHERE      civicrm_contact.is_deleted = 0 AND civicrm_event.id IN( {$eventID}) {$groupBy}
```

OR

```
LEFT JOIN  civicrm_participant ON ( civicrm_event.id = civicrm_participant.event_id
                       {$activeParticipantClause} AND civicrm_participant.is_test  = 0 )
JOIN civicrm_contact
          ON civicrm_participant.contact_id = civicrm_contact.id 
             AND civicrm_contact.is_deleted = 0
WHERE    civicrm_event.id IN( {$eventID}) {$groupBy}
```

Reply:
@pradpnayak I think "civicrm_participant.event_id is not indexed" is incorrect - the field looks to be indexed on my install

Also from my tests moving things between JOIN and WHERE like that probably won't change performance because mysql can still figure out that the contact_id is the best index to use in this case for the join on the contact table - but the only way to be sure is to test the queries on a biggish DB

Reply:
By civicrm_participant.event_id I meant its actually not joining the main table, Its a join made to subquery so thought it won't have indexing (sorry if i am wrong).

Reply:
@pradpnayak I think the answer is maybe - mysql optimisation engine figures some things out & not others so we'd need to test against mysql to see